### DISCUSSION TITLE
The proper way to intercept response at client

### DISCUSSION THREAD

**Discussion by addlistener** [trpc/trpc#3181] [Q&A]
I tried this but the proxy cannot be changed.
 
 The point is to make specific endpoint updating redux state ( the entity adapter ).

```
const original = tClient.user.get.query;

tClient.user.get.query = (...args) => {
  return original(...args).then((res) => {
    ... update redux state here ...
    return res;
  });
};
```

**Comment by addlistener** [comment]
I guess this is one way to change the interface 
<img width="739" alt="image" src="https://user-images.githubusercontent.com/4531670/202266066-99a31167-314e-400a-9e55-e77fd0fdd248.png">

However the typing benefits are lost this way.

**Comment by KATT** [comment]
Unsure what you're trying to achieve.... Maybe links works here?

**Reply by addlistener** [reply]
I'm trying to cache some response in frontend
